presto-hive/src/main/java/com/facebook/presto/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -0,0 +1,1970 @@
+                        String user = action.getUser();
+                        Path targetPath = new Path(table.getStorage().getLocation());
+                        Path currentPath = action.getData().getCurrentLocation().get();
+                        cleanUpTasks.add(new DirectoryCleanUpTask(DirectoryTaskKind.MOVE_BACKWARD, user, targetPath, false));
+                        if (!targetPath.equals(currentPath)) {
+                            asyncRename(hdfsEnvironment, renameExecutor, fileRenameCancelled, fileRenameFutures, user, currentPath, targetPath, action.getData().getFileNames().get());
+                        }
+                        break;
+                    }
+                    default:
+                        throw new IllegalStateException("Unknown action type");
+                }
+            }
+
+            for (Map.Entry<SchemaTableName, Map<List<String>, Action<PartitionAndMore>>> tableEntry : partitionActions.entrySet()) {
+                SchemaTableName schemaTableName = tableEntry.getKey();
+                PartitionAdder partitionAdder = new PartitionAdder(schemaTableName.getSchemaName(), schemaTableName.getTableName(), delegate, PARTITION_COMMIT_BATCH_SIZE);
+                partitionAdders.add(partitionAdder);
+                for (Map.Entry<List<String>, Action<PartitionAndMore>> partitionEntry : tableEntry.getValue().entrySet()) {
+                    List<String> partitionValues = partitionEntry.getKey();
+                    Action<PartitionAndMore> action = partitionEntry.getValue();
+                    switch (action.getType()) {
+                        case DROP:
+                            metastoreDeleteOperations.add(new IrreversibleMetastoreOperation(
+                                    format("drop partition %s.%s %s", schemaTableName, schemaTableName.getTableName(), partitionValues),
+                                    () -> delegate.dropPartition(schemaTableName.getSchemaName(), schemaTableName.getTableName(), partitionValues, true)));
+                            break;
+                        case ALTER: {
+                            String user = action.getUser();
+                            PartitionAndMore partitionAndMore = action.getData();
+                            Partition partition = partitionAndMore.getPartition();
+                            String targetLocation = partition.getStorage().getLocation();
+                            Optional<Partition> oldPartition = delegate.getPartition(schemaTableName.getSchemaName(), schemaTableName.getTableName(), partition.getValues());
+                            if (!oldPartition.isPresent()) {
+                                throw new PrestoException(
+                                        TRANSACTION_CONFLICT,
+                                        format("The partition you tried to modify was deleted by some one else. %s %s", schemaTableName.getTableName(), partition.getValues()));
```

Comment:
don't use personal words like `you` in error messages.  Use something like `Partition was modified in another transaction`